services/net/fileupload/FileUploadManager.cs

```diff
@@ -0,0 +1,74 @@
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using TNO.Ches;
+using TNO.Ches.Configuration;
+using TNO.Core.Exceptions;
+using TNO.Services.Managers;
+using TNO.API.Areas.Services.Models.Content;
+using System.Net.Http.Json;
+using TNO.Services.FileUpload.Config;
+namespace TNO.Services.FileUpload;
+using System.Net.Http;
+using System.Net.Http.Json;
+using Microsoft.AspNetCore.Mvc;
+using TNO.Core.Http;
+public class FileUploadManager : ServiceManager<FileUploadOptions>
+{
+    #region Variables
+    #endregion
+
+    #region Constructors
+    public FileUploadManager(
+        IApiService api,
+        IChesService chesService,
+        IOptions<ChesOptions> chesOptions,
+        IOptions<FileUploadOptions> options,
+        ILogger<FileUploadManager> logger)
+        : base(api, chesService, chesOptions, options, logger)
+    {
+    }
+    #endregion
+
+    #region Methods
+    public override async Task RunAsync()
+    {
+        if (this.State.Status != ServiceStatus.Running)
+        {
+            this.Logger.LogDebug("service is not running: '{Status}'", this.State.Status);
+            return;
+        }
+
+        try
+        {
+            this.Logger.LogInformation("start upload files to S3");
```

Comment:
Does this process run once?  Does it keep repeating?  Is the service supposed to stay running like all the other services?

Reply:
yes. it just run once.

Reply:
How does this shutdown after running?  If you're using the inherit services it will most likely keep running indefinitely.